### ISSUE TITLE
Allow creation of logical grouping of indexer types

### ISSUE THREAD

**Issue by binhex** [Jackett/Jackett#2185]
I have a "Enhancement", it would be really handy to be able to create logical groupings of indexers that the user has enabled, the logical group name could be defined by the user (with a default of something like groupXX) and once defined the user would then be able to add enabled indexers to the group which would result in the group members being aggregated together to create a logical url with link as per normal for torznab or potato feeds.

So onto the reasonings for wanting this:-

At the moment you need to copy and paste each individual link from each enabled indexer on jackett, which takes time and is error prone, and also means you need to keep adding when new indexers become available or removing when indexers disappear.
I am aware there is an "all" that you can use and whilst this will aggregate all of the enabled indexers it does then mean you might be needlessly trawling indexers that might not be suited for the metadata downloader app, as in sonarr processes tv series only so trawling movies is pointless.

An example of this would be a user creates 4 groups: usenet-movies, usenet-tv, torrent-movies, torrent-tv and then uses the Jackett logical group URL's for the relevant application, this would then allow the user to add/remove additional index sites whilst not having to maintain the list of enabled indexers on the downloader app. It would also mean the app would only process content that is relevant and thus would be much faster.

**Comment by Firelfy** [comment]
I would like this too. Also, possibly setup Torrents and Usenet as main groups with two subgroups each, one being movies and one being torrents.

So in case of using a tool that does movies and tv shows you can categorize based only on **[torrents]** or **[usenet]**, and in the case of sonarr you can choose the group **[torrents/tv shows]** or **[usenet/tv shows]** etc.

**Comment by kaso17** [comment]
dupe #921

**Comment by binhex** [comment]
i respectfully disagree about this being a dupe, issue #921 is about searching ALL jacket enabled indexers, this feature request is about having logical groupings of index sites and having the ability to search the jackett logical group url from your metadata downloader client (sonarr, radarr etc). i saw your comment @kaso17 about grouping so you obviously have the same notion as me, unless im mistaken though this never got implemented?

**Comment by kaso17** [comment]
reopened it.
actually @chibidev started to work on this via #1659
But as he's busy with other projects I wouldn't  expect any more progress on this for the foreseeable future. If someone want's to pick up hist work: pull requests are welcome.

**Comment by syco** [comment]
I guess this was originally addressed by the branch "feature/indexer-grouping".
Is there any plan to update it and merge it to the main branch?
Anyway.. thanks..

**Comment by Ansem93** [comment]
I hope this request will be implemented! It would be great to create groups based on lenguages and content for example!

**Comment by shagnome** [comment]
This would be a great feature, for the above reasons already explained.
Hoping it may be implemented in the future.

**Comment by Ansem93** [comment]
Yeah, I would rather stay some time without update for the tracker but see this feature implemented.

**Comment by knowair** [comment]
If this is to be accepted and integrated would it be possible to expand the Aggregate /all/  feature to allow similar multi-search functionality by giving indexers labels, ideally more than one label should be possible per indexer.

I envisage this being useful in a few ways, for example being able to group Public and Private trackers separately, or General and High Quality trackers. Perhaps, Standard TV and Anime. 

Replace the /all/ with /label-name/ and Jackett would only search the indexers with that label

i.e. if I have 5 indexers
i apply the label "public" to 3 of them and "private" to 2
the 3 with "public" 1 also has "tv" and 1 has "movies" the 3rd has both "tv" and "movies" labels
the 2 with "private" 1 also has "tv" and 1 has both "tv" and "movies" labels

so i could use 
/api/v2.0/indexers/private/results/torznab - Just for private trackers
or
/api/v2.0/indexers/movies/results/torznab - To only use the 3 indexers with the "movies" label regardless of them being private or not.

**Comment by STaRDoGG** [comment]
Ouch, over 3 years old, I guess we're sol. =/

**Comment by ilike2burnthing** [comment]
[v0.18.7](https://github.com/Jackett/Jackett/releases/tag/v0.18.7) - tags can now be added to indexers to group them together however you want.

You can then filter indexers for searches by these tags. Indexers can also be filtered by type (public, semi-private, and private). The API can also filter by language.

See more info here - https://github.com/Jackett/Jackett#filter-indexers
